python/ray/dataframe/dataframe.py

```diff
@@ -4852,15 +4848,18 @@ def del_helper(df, to_delete):
         # Cast cols as pd.Series as duplicate columns mean result may be
         # np.int64 or pd.Series
         col_parts_to_del = \
-            pd.Series(self._col_metadata[key, 'partition']).unique()
+            pd.Series(del_df['partition'].copy()).unique()
```

Comment:
Copy is necessary because `unique` will mutate the underlying data. If the underlying data comes from arrow sealed object, a "Buffer is read-only" exception will be raised.

Reply:
A lot of the caching and memoizing that Pandas does mutates the objects. In the future, hopefully we can get rid of the need for the copy.